### PR TITLE
imapd.c: add support for JMAPACCESS capability and response code

### PR DESCRIPTION
**Pull request by ksmurchison**
Implementation of 
[https://www.ietf.org/archive/id/draft-ietf-extra-jmapaccess-08.html](url)
without a JMAPACCESS capability (as discussed at IETF 119)

### CODE REVIEW

File: changes/next/imap_jmapaccess [hunk 1]
```diff
@@ -0,0 +1,19 @@
+
+Description:
+
+Adds support for IMAP JMAPACCESS response code (draft-ietf-extra-jmapaccess).
```
Comment (rsto) [inline changes/next/imap_jmapaccess:4]: Unless this gets tagged as `Do Not Merge` until that I-D draft is a standard, should this change log rather call out that this is a prototype implementation?

**Review by rsto — APPROVED** [review]
LGTM, with the caveat that this might rather not get merged in the main branch for now?

File: changes/next/imap_jmapaccess (hunk 1 above)
Comment (ksmurchison) [reply changes/next/imap_jmapaccess:4]: Switched PR to draft until IETF 199

**Comment by brong** [comment]
Yeah, let's deploy this on future and staff and make sure no clients barf!

File: imap/imapd.c [hunk 2]
```diff
@@ -416,7 +417,7 @@ static struct capa_struct base_capabilities[] = {
       { .statep = &imapd_idle_enabled }                       },
     { "IMAPSIEVE=",            0, /* not implemented */ { 0 } }, /* RFC 6785 */
     { "INPROGRESS",            CAPA_POSTAUTH,           { 0 } }, /* draft-ietf-extra-imap-inprogress */
-    { "JMAPACCESS",            0, /* not implemented */ { 0 } }, /* draft-ietf-extra-jmapaccess */
+    { "JMAPACCESS",            0, /* just a respcode */ { 0 } }, /* draft-ietf-extra-jmapaccess */
```
Comment (rsto) [inline imap/imapd.c:420]: typo here

**Review by rsto — APPROVED** [review]
LGTM but please fix the typo before merging

**Pull request merged**